testsuite/preview/basic/src/test/java/org/wildfly/test/preview/persistence/cdi/RequestScopedTestBean.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.test.preview.persistence.cdi;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import jakarta.persistence.Cache;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.metamodel.Metamodel;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.PersistenceUnitUtil;
+import jakarta.persistence.SchemaManager;
+import jakarta.transaction.Transactional;
+
+/**
+ * test bean
+ *
+ * @author Scott Marlow
+ */
+
+@RequestScoped
+public class RequestScopedTestBean {
+
+    @Produces
+    @PersistenceContext(unitName = "pu2")
+    private EntityManager emApplicationExistingProducer;
+
+    @Inject
+    @Pu1Qualifier
+    EntityManager em;
+
+    @Inject
+    @Pu1Qualifier
+    EntityManagerFactory emf;
+
+    @Inject
+    @Pu2Qualifier
+    EntityManagerFactory emf2;
+
+    @Inject
+    @Named("pu1")
+    EntityManagerFactory entityManagerFactoryByPuName;
+
+    @Inject
+    @Pu1Qualifier
+    CriteriaBuilder criteriaBuilder;
+
+    @Inject
+    @Pu1Qualifier
+    PersistenceUnitUtil persistenceUnitUtil;
+
+    @Inject
+    @Pu1Qualifier
+    Cache cache;
+
+    @Inject
+    @Pu1Qualifier
+    Metamodel metamodel;
+
+    @Inject
+    @Pu1Qualifier
+    SchemaManager schemaManager;
+
+    // The persistence unit with persistence unit hint "wildfly.jpa.default-unit" set to true should be the default persistence unit that is injected.
+    @Inject
+    EntityManagerFactory defaultEntityManagerFactory;
+
+    @Transactional
+    public Employee getEmployeeExpectNullResult(int id) {
+        return em.find(Employee.class, id);
+    }
+
+    @Transactional
+    public EntityManagerFactory entityManagerFactoryOfEntityManager() {
+        return em.getEntityManagerFactory();
+    }
+
+    public EntityManagerFactory injectedEntityManagerFactory() {
+            return emf;
```

Comment:
Nit: run a formatter on this line as it's off :)

Reply:
Hmm, looks like the random tabs insertion problem is back again.

Reply:
looking at https://bugs.kde.org/show_bug.cgi?id=484992 which was discussed before on https://wildfly.zulipchat.com/#narrow/channel/151151-general/topic/Fedora.2040.20.2B.20Intellij.20.2B.20alt-tab.20entering.20spaces.20into.20sources/with/459489796